### ISSUE TITLE
Also update the cart and TY page message

### ISSUE THREAD

**Issue by pdobrescu** [anttiviljami/woocommerce-custom-availability#1]
Firstly thanks for this great plugin! I was searching for something like this, but couldn't find it :)

I have one suggestion: it would be great if the plugin also updated the message that shows up in the cart and on the `thank you` page, because it currently shows `Available for backorder` in the cart and `Added on backorder` on the `thank you` page, which might be a bit misleading/confusing for the customers.

What do you think?

**Comment by anttiviljami** [comment]
Yes, definitely. Did you do the research to see if there are filters for that too in WooCommerce / your theme ?

**Comment by pdobrescu** [comment]
Wow, that was a fast reply :-D

Well, it's not out of the box, but it might be doable. The actual message in the cart gets displayed here:

https://github.com/woothemes/woocommerce/blob/master/templates/cart/cart.php#L92

There are some filters before, but I'm not sure if you can properly adjust the message. Maybe you can just add your information. Otherwise I can easily do this from my cart template I guess. 

As for the TY page, it loads the `order-details.php` template and most probably you can hook up here:

https://github.com/woothemes/woocommerce/blob/master/templates/order/order-details.php#L39

Thanks again!

**Comment by anttiviljami** [comment]
Hmm, looks like this is more of an issue with WooCommerce not being consistent with how the availability / backorder notification gets shown across the templates.

Looking at the links you provided, it's not immediately clear to me how you would efficiently filter the availability text at either template. There seems to be some logic involved there with the backorders that I'm not sure on how they should be taken to accont.

Let me know how you would like to proceed with this ! I'd be more than willing to take suggestions on this.

**Comment by BurlesonBrad** [comment]
@pdobrescu 
Are you wanting to change the actual text of `Available for backorder`?
Wanting to make it your own custom text **Universally**? 
_(implying a `/admin` text field needs to be added so that a store owner could change it on the backend)_
Wanting to make it your own custom text on a **Per-Product** basis?
_(implying a `/admin` text field needs to be added so that a store owner could change it on the backend)_

Dude with a little bit of work, @anttiviljami, you could totally take this plugin to the market and have people eating outta your hands :moneybag: :moneybag: :moneybag: :moneybag: :moneybag: 
![custom-availability-options](https://cloud.githubusercontent.com/assets/6208757/16784179/2cb404f2-484d-11e6-9155-e666bc829119.png)

Just a concept :wink:

**Comment by pdobrescu** [comment]
@BurlesonBrad just changing the text is pretty easy, that's not my point. I'm rather thinking about a **Per-Product** approach and your concept looks pretty neat. In a first phase it could be even easier than that, with just some check boxes with the places/templates where to amend and/or add the custom text.

@anttiviljami Indeed WooCommerce has become pretty huge and consistency is hard to keep at this level. On the other hand their developers are pretty responsive and requested filters/hooks are usually added. It definitely requires a deeper dive in the code for a proper solution. I'll try to get my hands a bit dirty sometime next week and see what comes out of it.

**Comment by pdobrescu** [comment]
@anttiviljami following the response from the WooCommerce team, a PR with the filters you need to properly making it work would be the way forward for solving this issue. More details here:

https://github.com/woothemes/woocommerce/issues/11505

Would that be something doable form your side?

**Comment by anttiviljami** [comment]
@pdobrescu Thanks for your work!

Yeah sure I can do that. I don't however have a use case for this right now, so it will probably have wait for my next project where I can bill a client for the work. 

Ofc. anyone is free to implement this feature if they need it for their own project and I'll happily merge it. :)

**Comment by raykai** [comment]
i would love to have this option too!

**Comment by raykai** [comment]
if you need $ to continue the work on this plugin you chould add this plugin on codecanyon https://codecanyon.net could get payed to continue to work on it with pro version that could have more options as long as its not 60$ and keep the free version on WP plugin list
